app/src/main/java/fr/acinq/phoenix/utils/Wallet.kt

```diff
@@ -133,17 +134,24 @@ object Wallet {
     val conf = HashMap<String, Any>()
 
     // electrum config
-    val electrumServer = Prefs.getElectrumServer(context)
-    if (!Strings.isNullOrEmpty(electrumServer)) {
+    val electrumServerAndProtocol = Prefs.getElectrumServer(context)
+    if (!Strings.isNullOrEmpty(electrumServerAndProtocol)) {
       try {
+        val ssl = when {
+          electrumServerAndProtocol.startsWith("ssl://") -> "strict" // custom server certificate must be valid
+          electrumServerAndProtocol.startsWith("tcp://") -> "off"
+          else -> throw IllegalArgumentException("invalid electrum uri")
```

Comment:
I think you can't throw here, otherwise all existing users that have electrum servers that didn't specify the protocol will hit that exception...maybe default to `ssl://`? Or do a kind of "migration" for existing users?

Reply:
Also, should not we support `tcp` only if the uri is a `.onion`? Otherwise, the app would support connecting to electrum servers with a self signed certificate.